visualization/src/svg.cpp

```diff
@@ -0,0 +1,67 @@
+#include <visualization/svg.hpp>
+
+namespace visualization::render {
+
+    SVGDimension::SVGDimension(double height, double width):
+        height(height), width(width) {}
+
+    double SVGDimension::get_width() const {
+        return width;
+    }
+
+    double SVGDimension::get_height() const {
+        return height;
+    }
+
+    SVGRender::SVGRender(const std::string name, const SVGDimension& dimension):
+        name(name), dimension(dimension) {}
+
+    SVGRender::~SVGRender() {}
+
+    void SVGRender::add_rectangle(const geometry::Rectangle& rectangle) {
+        XMLNode node = XMLNode::Builder{}.has_tag("rect")
+            .with_attribute("x", std::to_string(rectangle.get_center().get_x()))
+            .with_attribute("y", std::to_string(rectangle.get_center().get_y()))
+            .with_attribute("width", std::to_string(rectangle.get_width()))
+            .with_attribute("height", std::to_string(rectangle.get_height()))
+            .build();
+        
+        nodes.push_back(node);
+    }
+
+    void SVGRender::add_circle(const geometry::Circle& circle) {
+        XMLNode node = XMLNode::Builder{}.has_tag("circle")
+            .with_attribute("cx", std::to_string(circle.get_center().get_x()))
+            .with_attribute("cy", std::to_string(circle.get_center().get_y()))
+            .with_attribute("r", std::to_string(circle.get_radius()))
+            .build();
+        
+        nodes.push_back(node);
+    }
+
+    const std::string SVGRender::build_node(const XMLNode& node) const {
```

Comment:
For me, an `XMLNode` should be able to build itself (i.e. this should be moved to the XMLNode class)